core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,396 @@
+
+      // prepare transformed partition specs and caches
+      Map<Integer, PartitionSpec> transformedSpecs =
+          table().specs().values().stream()
+              .map(spec -> transformSpec(tableSchema(), spec))
+              .collect(Collectors.toMap(PartitionSpec::specId, spec -> spec));
+
+      LoadingCache<Integer, ResidualEvaluator> residualCache =
+          Caffeine.newBuilder()
+              .build(
+                  specId -> {
+                    PartitionSpec spec = transformedSpecs.get(specId);
+                    Expression filter =
+                        context().ignoreResiduals() ? Expressions.alwaysTrue() : filter();
+                    return ResidualEvaluator.of(spec, filter, isCaseSensitive());
+                  });
+      LoadingCache<Integer, String> specStringCache =
+          Caffeine.newBuilder()
+              .build(
+                  specId -> {
+                    PartitionSpec spec = transformedSpecs.get(specId);
+                    return PartitionSpecParser.toJson(spec);
+                  });
+
+      // iterate through delete manifests
+      CloseableIterable<ManifestFile> deleteManifests =
+          CloseableIterable.withNoopClose(snapshot().deleteManifests(tableOps().io()));
+      CloseableIterable<CloseableIterable<ScanTask>> results =
+          CloseableIterable.transform(
+              deleteManifests,
+              manifest -> {
+                // Filter partitions
+                CloseableIterable<ManifestEntry<DeleteFile>> deleteFileEntries =
+                    ManifestFiles.readDeleteManifest(manifest, tableOps().io(), transformedSpecs)
+                        .caseSensitive(isCaseSensitive())
+                        .select(scanColumns())
+                        .filterRows(rowFilter)
+                        .liveEntries();
+
+                // Filter delete file type
+                CloseableIterable<ManifestEntry<DeleteFile>> positionDeleteEntries =
+                    CloseableIterable.filter(
+                        deleteFileEntries,
+                        entry -> entry.file().content().equals(FileContent.POSITION_DELETES));
+
+                Types.StructType partitionType = Partitioning.partitionType(table());
+
+                return CloseableIterable.transform(
+                    positionDeleteEntries,
+                    entry -> {
+                      int specId = entry.file().specId();
+                      return new PositionDeleteScanTaskImpl(
+                          entry.file().copyWithoutStats(),
+                          schemaString,
+                          specStringCache.get(specId),
+                          residualCache.get(specId),
+                          partitionType);
+                    });
+              });
+
+      return new ParallelIterable<>(results, planExecutor());
+    }
+  }
+
+  /** Scan task for position delete files */
+  public interface PositionDeleteScanTask extends ContentScanTask<DeleteFile> {
```

Comment:
This should be a top-level class in `api` just like other public tasks.
Should we also call it `PositionDeletesScanTask` (plural)?

Reply:
Moved to top level